memphis/memphis.py

```diff
@@ -180,14 +183,47 @@ async def producer(self, station_name, producer_name, generate_random_suffix=Fal
                 createProducerReq, indent=2).encode('utf-8')
             err_msg = await self.broker_manager.request("$memphis_producer_creations", create_producer_req_bytes)
             err_msg = err_msg.data.decode("utf-8")
-
-            if err_msg != "":
+            err_msg = json.loads(err_msg)
+            if err_msg['error'] != "":
                 raise Exception(err_msg)
+            
+            self.t_schema_updates = await self.start_listen_for_schema_updates(station_name)
             return Producer(self, producer_name, station_name)
 
         except Exception as e:
             raise Exception(e)
 
+    async def get_msg_schema_updates(self, station_name_internal, iterable):
+        async for msg in iterable:
+            message = msg.data.decode("utf-8")
+            message = json.loads(message)
+            if message['UpdateType'] == 2:
+                data = {}
+            else:
+                data = message['init']
+            self.schema_updates_data[station_name_internal] =  data
+            self.schema_updates_data[station_name_internal]["producers_count"] = self.producers_count
```

Comment:
self is the connection and under 1 connection a lot of producers to totally different stations can live. so why aer you update the producers count of a specific station with the global counter